libraries/c_sdk/standard/ble/src/services/mqtt_ble/iot_ble_mqtt_serialize.c

```diff
@@ -85,575 +85,704 @@ static inline uint16_t _getNumPublishParams( const MQTTBLEPublishInfo_t * const
-        error = IOT_BLE_MESG_ENCODER.appendKeyValue( &publishMap, IOT_BLE_MQTT_PAYLOAD, data );
+        status = cbor_encode_text_string( &mapEncoder, IOT_BLE_MQTT_PAYLOAD, strlen( IOT_BLE_MQTT_PAYLOAD ) );
+
+        if( SERIALIZER_STATUS( status, pBuffer ) == true )
+        {
+            status = cbor_encode_byte_string( &mapEncoder, pPublishInfo->pPayload, pPublishInfo->payloadLength );
+        }
     }
 
-    if( _IS_VALID_SERIALIZER_RET( error, pBuffer ) )
+    if( pPublishInfo->qos != 0 )
     {
-        if( pPublishInfo->qos != 0 )
+        if( SERIALIZER_STATUS( status, pBuffer ) == true )
         {
-            data.type = IOT_SERIALIZER_SCALAR_SIGNED_INT;
-            data.value.u.signedInt = packetIdentifier;
-            error = IOT_BLE_MESG_ENCODER.appendKeyValue( &publishMap, IOT_BLE_MQTT_MESSAGE_ID, data );
+            status = cbor_encode_text_string( &mapEncoder, IOT_BLE_MQTT_MESSAGE_ID, strlen( IOT_BLE_MQTT_MESSAGE_ID ) );
+
+            if( SERIALIZER_STATUS( status, pBuffer ) == true )
+            {
+                status = cbor_encode_int( &mapEncoder, packetIdentifier );
+            }
         }
     }
 
-    if( _IS_VALID_SERIALIZER_RET( error, pBuffer ) )
+    if( SERIALIZER_STATUS( status, pBuffer ) == true )
     {
-        error = IOT_BLE_MESG_ENCODER.closeContainer( &encoderObj, &publishMap );
+        status = cbor_encoder_close_container( &encoder, &mapEncoder );
     }
 
-    if( _IS_VALID_SERIALIZER_RET( error, pBuffer ) )
+    if( pBuffer == NULL )
     {
-        if( pBuffer == NULL )
-        {
-            *pSize = IOT_BLE_MESG_ENCODER.getExtraBufferSizeNeeded( &encoderObj );
-        }
-        else
-        {
-            *pSize = IOT_BLE_MESG_ENCODER.getEncodedSize( &encoderObj, pBuffer );
-        }
+        *pLength = cbor_encoder_get_extra_bytes_needed( &encoder );
+    }
+    else
+    {
+        *pLength = cbor_encoder_get_buffer_size( &encoder, pBuffer );
+    }
 
-        IOT_BLE_MESG_ENCODER.destroy( &encoderObj );
-        error = IOT_SERIALIZER_SUCCESS;
+    if( SERIALIZER_STATUS( status, pBuffer ) == true )
+    {
+        result = CborNoError;
+    }
+    else
+    {
+        result = status;
     }
 
-    return error;
+    return result;
 }
 
-static IotSerializerError_t _serializePubAck( uint16_t packetIdentifier,
-                                              uint8_t * pBuffer,
-                                              size_t * pSize )
+static CborError prvSerializePubAck( uint16_t packetIdentifier,
+                                     uint8_t * pBuffer,
+                                     size_t * pLength )
 
 {
-    IotSerializerError_t error = IOT_SERIALIZER_SUCCESS;
-    IotSerializerEncoderObject_t encoderObj = IOT_SERIALIZER_ENCODER_CONTAINER_INITIALIZER_STREAM;
-    IotSerializerEncoderObject_t pubAckMap = IOT_SERIALIZER_ENCODER_CONTAINER_INITIALIZER_MAP;
-    IotSerializerScalarData_t data = { 0 };
+    size_t length = *pLength;
+    CborError status, result;
```

Comment:
Maintainability: status values should always be initialized to a value meaning _error_ or _failure_.  Same comment throughout this code.